mp4fpsmod: init at 0.27

makeding opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

makeding committed:
maintainers: add huggy

Sigmanificient commented on pkgs/by-name/mp/mp4fpsmod/package.nix:
```diff
@@ -0,0 +1,40 @@
+  stdenv,
+  fetchFromGitHub,
+  autoreconfHook,
+}@args:
```
```suggestion
}:
```

Sigmanificient commented on pkgs/by-name/mp/mp4fpsmod/package.nix:
```diff
@@ -0,0 +1,40 @@
+    owner = "nu774";
+    repo = "mp4fpsmod";
+    rev = "e2dd065012f4d2c7e42d4acdefee2ffdc50d3d86";
+    sha256 = "sha256-54pkjlvLLi4pLlQA/l+v4Mx5HlloR6GiB2GP71A0x/g=";
```
- #325892
```suggestion
    hash = "sha256-54pkjlvLLi4pLlQA/l+v4Mx5HlloR6GiB2GP71A0x/g=";
```

Sigmanificient commented on pkgs/by-name/mp/mp4fpsmod/package.nix:
```diff
@@ -0,0 +1,40 @@
+    ./bootstrap.sh
+  '';
+
+  meta = with lib; {
```
- #208242

```suggestion
  meta = {
```

Sigmanificient commented:
Missing a
`platforms = platforms.all;` -> `platforms = lib.platforms.all;`

makeding commented:
Thanks for the guidance of my code.
:)

FliegendeWurst commented on pkgs/by-name/mp/mp4fpsmod/package.nix:
```diff
@@ -0,0 +1,44 @@
+    '';
+    homepage = "https://github.com/nu774/mp4fpsmod";
+    license = with lib.licenses; [
+      # All files are distributed as Public Domain, except for the followings:
```
```suggestion
      # All files are distributed as Public Domain, except for the followings:
      publicDomain
```

FliegendeWurst commented on pkgs/by-name/mp/mp4fpsmod/package.nix:
```diff
@@ -0,0 +1,44 @@
+      bsd2 # FreeBSD CVS
+    ];
+    platforms = lib.platforms.all;
+    maintainers = with lib.maintainers; [ huggy ];
```
```suggestion
    maintainers = with lib.maintainers; [ huggy ];
    mainProgram = "mp4fpsmod";
```

FliegendeWurst commented:
Please squash all package-related commits.

Sigmanificient commented on pkgs/by-name/mp/mp4fpsmod/package.nix:
```diff
@@ -0,0 +1,46 @@
+      delaying audio tracks, executing DTS compression, extracting
+      time codes of mp4.
+    '';
+    homepage = "https://github.com/nu774/mp4fpsmod";
```
```suggestion
    interhit (src.meta) homepage;
```

Guanran928 reviewed (approved):
built on x86_64-linux: `/nix/store/sz0wwccxpqkzdvcs6gr705558b9yk6r0-mp4fpsmod-0.27-unstable-2023-12-30`

FliegendeWurst commented on pkgs/by-name/mp/mp4fpsmod/package.nix:
```diff
@@ -0,0 +1,46 @@
+      boost # Boost
+      bsd2 # FreeBSD CVS
+    ];
+    platforms = platforms.all;
```
```suggestion
    platforms = platforms.unix;
```
Does not cross-compile to Windows: `error: #error Please define `_WIN32_WINNT` to at least Windows 7.`

makeding committed:
mp4fpsmod: init at 0.27

Update pkgs/by-name/mp/mp4fpsmod/package.nix

Co-authored-by: Yohann Boniface <edhyjox@gmail.com>

Update pkgs/by-name/mp/mp4fpsmod/package.nix

Co-authored-by: Yohann Boniface <edhyjox@gmail.com>

mp4fpsmod: #333636 platforms.all -> lib.platforms.all

Update pkgs/by-name/mp/mp4fpsmod/package.nix

Co-authored-by: Arne Keller <2012gdwu+github@posteo.de>

platforms.all to platforms.unix(does not support windows)

Co-authored-by: Arne Keller <2012gdwu+github@posteo.de>
